test/src/tests/generated/model-flattening/models.go

```diff
@@ -107,31 +243,100 @@ type ProductWrapper struct {
 	*WrappedProduct `json:"property,omitempty"`
 }
 
+// UnmarshalJSON is the custom unmarshaler for ProductWrapper struct.
+func (pw *ProductWrapper) UnmarshalJSON(body []byte) error {
+	var m map[string]*json.RawMessage
```

Comment:
Why does this one need a custom unmarshaller?

Reply:
We need this because there is a an embedded type there and there is the risk of using the embedded type's unmarshaller.